src/Features/CSharp/Portable/CodeLens/DisplayInfoCSharpServices.cs

```diff
@@ -0,0 +1,348 @@
+                   node.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia);
+        }
+
+        /// <summary>
+        /// Returns the node that should be displayed
+        /// </summary>
+        public SyntaxNode GetDisplayNode(SyntaxNode node)
+        {
+            while (true)
+            {
+                switch (node.Kind())
+                {
+                    // LocalDeclarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.LocalDeclarationStatement:
+                        var localDeclarationNode = (LocalDeclarationStatementSyntax) node;
+                        node = localDeclarationNode.Declaration.Variables.First();
+                        continue;
+
+                        // Field and event declarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.FieldDeclaration:
+                    case SyntaxKind.EventFieldDeclaration:
+                        var fieldNode = (BaseFieldDeclarationSyntax) node;
+                        node = fieldNode.Declaration.Variables.First();
+                        continue;
+
+                        // Variable is a field without access modifier. Parent is FieldDeclaration
+                    case SyntaxKind.VariableDeclaration:
+                        node = node.Parent;
+                        continue;
+
+                        // Built in types   
+                    case SyntaxKind.PredefinedType:
+                        node = node.Parent;
+                        continue;
+
+                    case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    case SyntaxKind.SingleLineDocumentationCommentTrivia:
+                        // For DocumentationCommentTrivia node, node.Parent is null. Obtain parent through ParentTrivia.Token
+                        if (node.IsStructuredTrivia)
+                        {
+                            var structuredTriviaSyntax = (StructuredTriviaSyntax) node;
+                            node = structuredTriviaSyntax.ParentTrivia.Token.Parent;
+                            continue;
+                        }
+                        return null;
+
+                    default:
+                        return node;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the DisplayName for the given node.
+        /// </summary>
+        public string GetDisplayName(
+            SemanticModel semanticModel,
+            SyntaxNode node,
+            DisplayFormat displayFormat)
+        {
+            if (node == null)
+            {
+                return CSharpFeaturesResources.unknown_value;
+            }
+
+            SymbolDisplayFormat symbolDisplayFormat = DefaultFormat;
+            if (displayFormat == DisplayFormat.Short)
+            {
+                symbolDisplayFormat = ShortFormat;
+            }
+
+            string displayName;
```

Comment:
In this case, the variable is assigned across a switch statement and then returned. The code could be restructured so that each case declares displayName and then returns at the end of the case but that kinda seems overkill.

---
In reply to: [73584236](https://github.com/dotnet/roslyn/pull/12930#discussion_r73584236) [](ancestors = 73584236)